### PR TITLE
Migrate crypto/* to proptest

### PR DESCRIPTION
**Pull request by z2trillion**
- [ ] Tag the PR with `wip` while in development.
- [ ] Assign yourself as to the PR
- [ ] Assign relevant labels such as `bug`, `enhancement`.
- [ ] Request reviews if the PR is large, complex or you would like an extra pair of eyes to go over it.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [ ] Add new entries to the `Changelog.md`.
- [ ] Update version numbers as needed.

Semver: https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md

https://semver.org/

### CODE REVIEW

File: crypto/stark/examples/pedersen_merkle/component.rs [hunk 1]
```diff
@@ -367,16 +369,18 @@ mod test {
...
+        fn test_pedersen_merkle(seed: u64) {
+            let mut rng = Xoshiro256PlusPlus::seed_from_u64(seed);
+            let size = 1 << Uniform::from(0..4).sample(&mut rng);
+            let witness = Witness {
+                directions: (0..size).map(|_| rng.gen()).collect(),
+                path:       (0..size).map(|_| rng.gen()).collect(),
+            };
+            let claim = Claim::from_leaf_witness(rng.gen(), &witness);
+            let component = pedersen_merkle(&claim, &witness);
+            prop_assert!(component.check());
```
Comment (Recmo) [inline crypto/stark/examples/pedersen_merkle/component.rs]: This no longer needs the Rng and could generate the test case using `Strategy`.

Comment (Recmo) [reply crypto/stark/examples/pedersen_merkle/component.rs]: We could leave a `// TODO` for it.

File: Codeguidelines.md [hunk 2]
```diff
@@ -32,8 +32,8 @@ We follow the
 
 Except,
 
-* Unit tests are in the same file as the code covered and not in separate files.
-* We use `quickcheck` for property based testing.
+-   Unit tests are in the same file as the code covered and not in separate files.
+-   We use `proptest` for property based testing.
```
Comment (Recmo) [inline Codeguidelines.md]: Why change from `*` to `-`?

File: crypto/elliptic-curve-crypto/src/lib.rs [hunk 3]
```diff
@@ -198,10 +196,12 @@ mod tests {
...
-        verify(&digest, &signature, &public_key)
-    }
+    proptest!(
+        #[test]
+        fn test_ecdsa(digest: ScalarFieldElement, private_key: ScalarFieldElement) {
+            let public_key = private_to_public(&private_key);
+            let signature = sign(&digest, &private_key);
+            prop_assert!(verify(&digest, &signature, &public_key));
+        }
+    );
```
Comment (Recmo) [inline crypto/elliptic-curve-crypto/src/lib.rs]: How do we feel about this vs using the closure notation for proptest?

Comment (z2trillion) [reply crypto/elliptic-curve-crypto/src/lib.rs]: I like the closure notation. I'm going to switch everything over in another PR.

**Pull request merged**